app/controllers/BulkRequestReceivedController.scala

```diff
@@ -39,24 +39,22 @@ class BulkRequestReceivedController @Inject()(authAction: AuthAction,
   def get = authAction.async {
       implicit request => {
         val link = request.linkId
-
         Logger.debug(s"[BulkRequestReceivedController][get][GET] : ${request.body}")
         sessionService.fetchGmpBulkSession().flatMap {
           case Some(session) if session.callBackData.isDefined =>
             val callbackData = session.callBackData.get
-
             bulkRequestCreationService.createBulkRequest(callbackData.collection, callbackData.id, session.emailAddress.getOrElse(""), session.reference.getOrElse("")) match {
```

Comment:
Consider refactoring createBulkRequest to work with standard paradigm where Left contains the error cases and Right is the happy path.